docs: move docs branch sections from readme to contrib guide (#1902)
keptn/lifecycle-toolkit: docs/CONTRIBUTING.md

This page reflects the current development status of the documentation.
It will be built regularly and can be easily accessed.

- build: on each push to `main` with documentation changes
- build-environment: main
- config folder: [main](./config/staging/)

It should allow bleeding-edge users and contributors to see the current state and help with debugging etc.

### Official documentation (Production)

This documentation set contains all released versions of KLT and is stored in an orphaned branch called `page`.

- build: on each push to `page` with documentation changes
- build-environment: production
- config folder: [production](./config/production/)

## FAQs

- Who/how do these changes get reviewed/approved/merged?
```
Like every normal PR - if the changes are approved by the maintainer, it will get merged.
```

- Which branch should I consider when contributing?
```
New writing goes to the main branch and can be viewed on the Releases -> development dropdown menu
```

- When we push the main branch to production, does it overwrite changes that were only pushed to page?
```
No, because a new version is generated - hence old page branch will be copied over a “version” folder

> ```

Reviewer: I would put the questions into the paragraphs that talk about the related sections instead of doing an FAQ here.
- how do changes get reviewed/approved: i don't think this question is really necessary here, it's explained in the PR creation paragraphs already.
- which branch should i consider: this could be added to "choosing the correct branch when contributing"
- are changes overwritten: also for this, i would add a sentence to the "choosing the correct branch when contributing" section

Author: Done :+1:

   ```console
   git branch -d <branch-name>
   ```
   
## Publishing

We are using Netlify to publish our pages.
There are 3 different types of publication:

1. pull request previews
2. development documentation aka staging (build of `main` branch) - [link](https://main.lifecycle.keptn.sh)
3. official documentation aka production (build of `page` branch) - [link](https://lifecycle.keptn.sh)

> Within the navigation bar, we do have version links pointing to the different publications - if it makes sense.

Reviewer: ```suggestion
Within the navigation bar, we have version links pointing to the different publications - if it makes sense.
```

   git branch -d <branch-name>
   ```

## Choosing the correct branch when contributing

New writing goes to the `main` branch and can be viewed on the Releases -> development dropdown menu.
We have staging and production levels for our documentation which are as follows:

> 1. Development documentation aka staging (build of `main` branch) - [link](https://main.lifecycle.keptn.sh)
> 2. Official documentation aka production (build of `page` branch) - [link](https://lifecycle.keptn.sh)

Reviewer 2: ```suggestion
* Official documentation aka production (build of `page` branch)
  - [link](https://lifecycle.keptn.sh)
* Development documentation aka staging (build of `main` branch) - [link](https://main.lifecycle.keptn.sh)
* Documentation for earlier releases
```
Let's list the production docs first, and it is not a sequential list

Let's take a look what happens when your changes are merged in `main` and `page` branch respectively.

> ### Development documentation (Staging)

Reviewer 2: ```suggestion
## Development documentation (Staging)
```

> This page reflects the current development status of the documentation.

Reviewer 2: ```suggestion
This page contains the documentation being staged for the next Keptn release.
It contains information about new and changed features and functionality
as well as general documentation improvements.

> It will be built regularly and can be easily accessed.

Reviewer 2: ```suggestion
It is built regularly and can be easily accessed from the `Releases` tab on the documentation site.
```

> * build: on each push to `main` with documentation changes

Reviewer 2: ```suggestion
* build: on each push to `main` with documentation changes
   from a user's local branch, from the github editor, or from codespaces
```

* build-environment: main
* config folder: [main](./config/staging/)

> It should allow bleeding-edge users and contributors to see the current state and help with debugging etc.

Reviewer 2: ```suggestion
This is the staged version of the documentation for the next Keptn release.
PRs generated from a user's local branch, from the github editor, or from codespaces
are merged into this branch.

This branch is promoted to `latest` when a new Keptn version is released.
On rare occasions, a PR that contains critical improvements to the documentation
may be silently promoted to `latest` without a software release.
```

> ### Official documentation (Production)

Reviewer 2: ```suggestion
## Latest -- Official documentation (Production)
```
This should be the first sub-section, preceding info about the development docs

> This documentation set contains all released versions of KLT and is stored in an orphaned branch called `page`.

Reviewer 2: ```suggestion
This documentation set contains documentation for the current KLT release
and is stored in an orphaned branch called `page`.
When a new KLT version is released,
the content of the `development` branch is rolled to this branch.
Note also that doc modifications made through the
"Edit this page" facility are merged directly into this branch.

This uses the `latest` label so that links to a doc page
remain valid across software and documentation updates. 
```

* build: on each push to `page` with documentation changes
* build-environment: production
* config folder: [production](./config/production/)

A new version is generated when we push the `main` branch to production to release a new version of the docs page.

> This means, that the content of the old version on the `page` branch will be copied over to a `docs-<version>` folder and the new version will be pushed into the `docs` folder.

Reviewer 3: ```suggestion
This means, that the content of the old version on the `page` branch will be copied over
to a `docs-<version>` folder and the new version will be pushed into the `docs` folder.
```

Reviewer 3: for making the linting happy :)

   git branch -d <branch-name>
   ```

## Choosing the correct branch when contributing

New writing goes to the `main` branch and can be viewed on the Releases -> development dropdown menu.
We have staging and production levels for our documentation which are as follows:

> * **Latest:** official documentation of the current Keptn release
>    - [link](https://lifecycle.keptn.sh):
>       This is the build of the `page` branch.
>    
> * **Development:** documentation being staged for the next Keptn release
>    - [link](https://main.lifecycle.keptn.sh):
>    This is the latest build of the `main` branch.

Reviewer 4: ```suggestion
* **Latest:** official documentation of the current Keptn release
  * [link](https://lifecycle.keptn.sh):
      This is the build of the `page` branch.

* **Development:** documentation being staged for the next Keptn release
  * [link](https://main.lifecycle.keptn.sh):
   This is the latest build of the `main` branch.
```

* **Previous versions:** documentation for earlier releases.
   These are listed at [link](https://github.com/keptn/lifecycle-toolkit/tree/page/docs/content/en).
  
* **Contribute:** current version of the "Contribute" guide
   that is available from a tab on the documentation site.

Let's take a look what happens when your changes are merged in `main` and `page` branch respectively.

## Latest -- Official documentation (Production)

> This set of documentation pertains to the latest KLT release and resides within an
> isolated branch known as `page`. When a new version of KLT is launched, the contents of the `development`
> branch are rolled into this branch. Additionally, it's important to recognize that any
> document changes made using the "Edit this page" feature are seamlessly integrated into this branch.

Reviewer 4: ```suggestion
This set of documentation pertains to the latest KLT release and resides within an
isolated branch known as `page`.
When a new version of KLT is launched, the contents of the `development`
branch are rolled into this branch.
Additionally, it's important to recognize that any
document changes made using the "Edit this page" feature are seamlessly integrated into this branch.
```

This uses the `latest` label so that links to a doc page

> remain valid across software and documentation updates. 

Reviewer 4: ```suggestion
remain valid across software and documentation updates.
```

to a `docs-<version>` folder and the new version will be pushed into the `docs` folder.
This way, no changes or older versions get overwritten.

## Development documentation (Staging)

This page contains the documentation being staged for the next Keptn release.
It contains information about new and changed features and functionality
as well as general documentation improvements.
It is built regularly and can be easily accessed from the `Releases` tab on the documentation site.

* build: on each push to `main` with documentation changes
   from a user's local branch, from the github editor, or from codespaces
* build-environment: main
* config folder: [main](./config/staging/)

This version represents the pre-release iteration of the documentation for the upcoming Keptn release.
Pull requests originating from a user's local branch, the GitHub editor, or codespaces are merged into this branch.

When a new Keptn version is officially launched, this branch is elevated to the status of `latest`.
In exceptional cases, a pull request that includes vital documentation enhancements may be discreetly
advanced to `latest` without the need for a software release.

## Previous Versions

> Keptn documentation is versioned. By default, the version for the current Keptn release
> is displayed on the documentation page but users can select other versions from the Releases tab. 
> The previous versions of the Keptn Documentation are available [here](https://github.com/keptn/lifecycle-toolkit/tree/page/docs/content/en).
> 

Reviewer 4: ```suggestion
Keptn documentation is versioned.
By default, the version for the current Keptn release
is displayed on the documentation page but users can select other versions from the Releases tab.
The previous versions of the Keptn Documentation are available [here](https://github.com/keptn/lifecycle-toolkit/tree/page/docs/content/en).
```

- build-environment: production
- config folder: [production](./config/production/)

> A new version is generated when we push the `main` branch to production - hence old `page` branch will be copied over a “version” folder and it doesn't overwrite changes that were only pushed to `page`.

Reviewer: ```suggestion
A new version is generated when we push the `main` branch to production to release a new version of the docs page.
This means, that the content of the old version on the `page` branch will be copied over to a `docs-<version>` folder and the new version will be pushed into the `docs` folder.
This way, no changes or older versions get overwritten.
```

Author: Ohh this sounds much better :+1:
